documentation.lisp

```diff
@@ -33,6 +33,19 @@ Supported on:
 
 Defaults to MOST-NEGATIVE-SHORT-FLOAT on unsupported implementations.")
 
+  (variable short-float-nan
+    "A positive quiet NaN value with zero-cleared payload for short-floats.
+This value should NOT be used for comparison with a float value for the pupose of testing NaN
+because NaN has unused bits (called payload) which are explicitly allowed to store additional information.
+Comparing this value with other NaN representations via = will fail.
+To test if a number is a NaN, use FLOAT-NAN-P.
+
+Supported on:
+* CCL
+* SBCL
```

Comment:
According to the code it looks like a lot more implementations are supported?

Also, I'm not sure about the default value. I feel like on unsupported implementations it should be a symbol macro that errors.

Reply:
it looks as if it supported, but the value is same as the default infinite positive value --- so basically same as being unsupported. someone with experience with those implementations may make them supported.

re: default value, that was a tough choice, but it was chosen because the representation is the closes nan value.
infinity: 
01111111100000000000000000000000
quiet nan:
01111111110000000000000000000000

behavior wise, if you care about the correctness of `nan = infinity`, then `infinity = most positive float` is also incorrect... it is broken anyways. The advantage over signaling errors is that the code using nan is still runnable.

Reply:
Couldn't you use bits-single-float etc. to construct the NaNs on a few other implementations?

Reply:
Oh, I was not looking at those functions, yes I should definitely use them. Will update the PR, thanks

Reply:
But given that bits-long-float / bits-short-float are missing, we should decide them

Reply:
I also noticed that `#+(and long-float sbcl)` should use a 128-bit float...

Reply:
early-float.lisp
```
#+(not long-float)
(defconstant long-float-negative-infinity double-float-negative-infinity)
#+(and long-float x86)
(defconstant long-float-negative-infinity
  (long-from-bits 1 (1+ sb-vm:long-float-normal-exponent-max)
                  (ash sb-vm:long-float-hidden-bit 32)))
```